service/src/main/java/uk/gov/hmcts/reform/fpl/service/OthersService.java

```diff
@@ -31,6 +40,61 @@ public String buildOthersLabel(Others others) {
         return sb.toString();
     }
 
+    public Others prepareOthers(CaseData caseData) {
+        List<Element<Other>> others = new ArrayList<>();
+
+        caseData.getAllOthers().forEach(element -> {
+            if (element.getValue().containsConfidentialDetails()) {
+                Element<Other> confidentialOther = findConfidentialOther(caseData.getConfidentialOthers(), element);
+
+                others.add(element(element.getId(), addConfidentialDetails(confidentialOther, element)));
+            } else {
+                others.add(element);
+            }
+        });
+
+        return Others.builder()
+            .firstOther(getFirstOther(caseData.getConfidentialOthers(), others))
+            .additionalOthers(getAdditionalOthers(others))
+            .build();
+    }
+
+    private Element<Other> findConfidentialOther(List<Element<Other>> confidentialOthers, Element<Other> element) {
```

Comment:
We can use the similar method in confidential details service called getItemToAdd

Reply:
moved it to Helper class